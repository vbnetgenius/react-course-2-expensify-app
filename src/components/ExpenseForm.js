import React from 'react';
import { connect } from 'react-redux';

export class ExpenseForm extends React.Component {

    state = {
        note: this.props.expense ? this.props.expense.note : '',
        amount: this.props.expense ? this.props.expense.amount : '',
        description: this.props.expense ? this.props.expense.description : '',
        error: undefined,    
    };

    handleAddExpense = (e) => {
        e.preventDefault();  
        
        if (!this.state.note || !this.state.amount || !this.state.description) {
            this.setState(() => ({
                error: 'Please fill out form fields',
            }));
        } else {
            
            this.setState(() => ({
                error: '',
            }));
                    
            this.props.onSubmit({
                note: this.state.note,
                amount: this.state.amount,
                description: this.state.description,                
            });

        }

    };

    handleChange = (e) => {
        this.setState((prevState) => {
            return {
                [e.target.name]: e.target.value,
            };
        });
    };

    render() {
        return <div>

            { this.state.error && <p>{this.state.error}</p> }

            <form onSubmit={this.handleAddExpense}>
                <input name="description" type="text" placeholder="Description" autoFocus={true} onChange={this.handleChange} />
                <input name="amount" type="text" placeholder="Amount" onChange={this.handleChange} />            
                <textarea name="note" cols="30" rows="10" onChange={this.handleChange}></textarea>           
                <button>Add Expense</button>
            </form>

        </div>
    }

}

export default connect()(ExpenseForm);