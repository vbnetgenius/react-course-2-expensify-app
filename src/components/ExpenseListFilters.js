import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter } from './../actions/filters';

const ExpenseListFilters = (props) => (
    <div>
        <input type="text" value={props.text} onChange={(e) => { props.dispatch(setTextFilter({text: e.target.value})) }} />
    </div>
);

const mapStateToProps = (state) => {
    return {
        text: state.filters.text,
    };
};

export default connect(mapStateToProps)(ExpenseListFilters);