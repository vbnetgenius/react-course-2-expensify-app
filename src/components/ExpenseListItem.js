import React from 'react';
import { connect } from 'react-redux';
import { removeExpense } from '../actions/expenses';

export const ExpenseListItem = (props) => {
    const { id, description, amount, createdAt } = props;
    return ( 
        <div>
            <h1>{description}</h1>
            <p>{amount} : {createdAt}</p>
            <button onClick={() => { props.dispatch( removeExpense( {id} ) ); } }>Delete</button>
        </div>
     );
}
 
export default connect()(ExpenseListItem);