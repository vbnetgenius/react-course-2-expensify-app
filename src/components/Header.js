import React from 'react';
import { NavLink } from 'react-router-dom';


const Header = (props) => (
    <div>
        <h1>Siavash</h1>
        <NavLink to="/dashboard" activeClassName="bold">Dashboard</NavLink>
        <NavLink to="/profile" activeClassName="bold">Profile</NavLink>
        <NavLink to="/users" activeClassName="bold">Users</NavLink>
        <NavLink to="/about" activeClassName="bold">About</NavLink>
    </div>
);

export default Header;