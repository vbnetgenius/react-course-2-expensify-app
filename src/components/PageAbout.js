import React from 'react';

const PageAbout = () => (
    <div>About Page</div>
);

export default PageAbout;