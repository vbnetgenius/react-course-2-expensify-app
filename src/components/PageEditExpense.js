import React from 'react';

const PageEditExpense = (props) => {
    const expenseID = props.match.params.id;
    return <div>Editing: {expenseID}</div>;
} 

export default PageEditExpense;