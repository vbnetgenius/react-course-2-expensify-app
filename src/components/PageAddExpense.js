import React from 'react';
import { connect } from 'react-redux';
import { addExpense } from '../actions/expenses';
import ExpenseForm from './ExpenseForm';

const PageAddExpense = (props) => (
    <React.Fragment>
        <h1>Add Expense</h1>
        <ExpenseForm onSubmit={(payload) => { 
            props.dispatch(addExpense({
                ...payload
            }));  
         }} />
    </React.Fragment>
);

export default connect()(PageAddExpense);