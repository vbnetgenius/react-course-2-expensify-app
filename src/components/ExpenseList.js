import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';
import { v4 as uuid } from 'uuid';

export const ExpenseList = (props) => {    
    return <div>
    {
        props.expenses.length === 0 ? (
            <p>No expenses</p>
        ) : (
            props.expenses.map(e => <ExpenseListItem key={uuid()} {...e} />)
        )
    }
</div>;
}
 
const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters),
    };
};

export default connect(mapStateToProps)(ExpenseList);