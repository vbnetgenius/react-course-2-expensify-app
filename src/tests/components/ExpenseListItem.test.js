import React from 'react';
import { shallow } from 'enzyme';
import { Expense } from './../fixtures/expense';
import toJSON from 'enzyme-to-json';
import { ExpenseListItem } from '../../components/ExpenseListItem';

test('should render ExpenseListItem with fixture', () => {
    const wrapper = shallow(<ExpenseListItem {...Expense} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
});