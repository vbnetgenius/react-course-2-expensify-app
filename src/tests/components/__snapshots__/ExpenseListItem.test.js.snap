// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem with fixture 1`] = `
<div>
  <h1>
    خرج و مخارج منزل
  </h1>
  <p>
    155000
     : 
    87634435
  </p>
  <button
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
